interactor/book.go

```
@@ -53,3 +69,51 @@ func (b *Book) Delete(*model.Book) error {
 
 	return nil
 }
+
+func toBook(book *openlibrarydto.Book) (bm *model.Book) {
+	isbn10 := ""
+	if book.Identifier.ISBN10 != nil {
+		isbn10 = book.Identifier.ISBN10[0]
+	}
+	isbn13 := ""
+	if book.Identifier.ISBN13 != nil {
+		isbn13 = book.Identifier.ISBN13[0]
+	}
+
+	CoverId := ""
+	if book.Cover.Url != "" {
+		part1 := strings.Split(book.Cover.Url, "/")[5]
+		part2 := strings.Split(part1, ".")[0]
+		CoverId = strings.Split(part2, "-")[0]
+	}
+	libraryId := ""
+	if book.Identifier.Openlibrary != nil {
+		libraryId = book.Identifier.Openlibrary[0]
+	}
+	author := ""
+	if book.Author != nil {
+		author = book.Author[0].Name
+	}
+
+	year := 0
+	var err error
+	yearString := yearRgx.FindString(book.Year)
+	if yearString != "" {
+		year, err = strconv.Atoi(yearString)
+		if err != nil {
+			fmt.Println("error while converting year from string to int", err)
+			return nil
```

we should return this error instead of logging